python/solid_dmft/dmft_tools/solver.py

```diff
@@ -681,10 +682,20 @@ def _create_hartree_solver(self):
         from hartree_fock.impurity import ImpuritySolver as hartree_solver
 
         gf_struct = self.sum_k.gf_struct_solver_list[self.icrsh]
+        n_orb = self.sum_k.corr_shells[self.icrsh]['dim']
+
+
         # Construct the triqs_solver instances
         triqs_solver = hartree_solver(beta=self.general_params['beta'], gf_struct=gf_struct,
                                       n_iw=self.general_params['n_iw'], force_real=self.solver_params['force_real'],
-                                      symmetries=[self._make_spin_equal])
+                                      symmetries=[self._make_spin_equal],
+                                      U= self.general_params['U'][self.icrsh],
+                                      J= self.general_params['J'][self.icrsh],
+                                      n_orb = n_orb
+                                      )
+
+        # Give dc information to the solver in order to customize DC calculation
+        triqs_solver._interface_to_solid_dmft(self.general_params, self.advanced_params, self.icrsh)
```

Comment:
Why is the interface function private (starts with an _)?

Reply:
I think if this function is doing something specific to solid_dmft it should be moved out of the solver somewhere. See my comment in the Hartree solver PR.

Reply:
Moved definition to solid_dmft